---
title: "DNA Variant Analysis with R Bioconductor"
output: html_notebook
---

<br/>

#### *Objective 1. Examine the header and metadata of a Variant Call Format (VCF) file*

<br/>

##### Get a raw VCF file to practice with  


We will read a file that is already stored in your project directory. You can retrieve this directory on your system using the `here()` function.  

```{r}
proj_dir <- here() ## should be the project directory even if getwd() shows something else

vcf_file <- paste0(proj_dir, "/Data/NA06985_17.vcf.gz") 
vcf_file
```


Optional: Make a copy of the data and then decompress it so we can view it in a text editor. If you do _not_ have the `R.utils` package, you could also manually double click on the .gz file to decompress it.  
```{r message=FALSE}
R.utils::gunzip(vcf_file, skip=TRUE, remove=FALSE)
```


If you open this in a text editor, you'll see a large header. Yikes!

<br/>

##### Read the VCF into R

We can use the `scanVcfHeader()` function to get all of this information stored in an object `hdr` that we can retrieve values from, such as the file metadata, data sources and other genotype information.  

```{r}
?scanVcfHeader

hdr <- scanVcfHeader(vcf_file)
hdr
```

<br/>

The __metadata__:

```{r}
meta(hdr)
```

<br/>

The __information__:
```{r}
info(hdr)
```
<br/>

The __genotypes__:
```{r}
geno(hdr)
```

<br/>

The __slots__:
```{r}
slotNames(hdr)
```

Access slots with `@`:

```{r}
hdr@samples ## slot
```
