---
title: "DNA Variant Analysis with R Bioconductor"
output: html_notebook
---

<br/>

#### *Objective 6: Export variants to a Browser Extensible Data (BED) file and view in UCSC browser*

<br/>

Let's subset the variants on non-TRP genes so we can visualize where these are in the UCSC browser. Recall from the challenge in Objective 4, we created `other_ids`.  
```{r}
table("gene loc:"=all_df$LOCATION, "ENTREZID:"=all_df$GENEID)

other_ids
```

<br/>

We have 2 variants in SHPK with ENTREZID 23729 so let's select those from `all_vars`.
```{r warning=FALSE}
shpk_vars <- all_vars[!is.na(all_vars$GENEID=="23729")]
shpk_vars
```

<br/>

Now recall that we can get the genotypes for these variants with the `ScanVcfParam()` and `readVcf()` functions.
```{r}
shpk_params <- ScanVcfParam(info="DP", geno=c("GT", "cPd"), which=shpk_vars)
shpk_params
shpk_vcf <- readVcf(vcf_file, "hg19", shpk_params)
shpk_vcf
rowRanges(shpk_vcf)
```

<br/> 

So we see that the first of these is a single nucleotide polymorphism (SNP) and the second one is a deletion of GAGA.  

The ENSEMBL website has a nice summary of small variant types found in VCFs   (http://uswest.ensembl.org/info/genome/variation/index.html):  


<br/>

In order to view these in the UCSC browser, we need to set up a special GRanges object for this with just the chromosome (back to "chr"), ranges, strand, and variant types.
```{r}
?GRangesForUCSCGenome

shpk_vars@ranges
shpk_vars@strand

shpk_track <- GRangesForUCSCGenome(genome = "hg19", 
    chrom = "chr17", 
    ranges = shpk_vars@ranges,
    strand = shpk_vars@strand)

shpk_track
```

<br/>
Now we can save the BED file.
```{r}
export(shpk_track, paste0(proj_dir, "/Data/shpk.bed"))
```

<br/>
Finally, let's view the second one (deletion) in the UCSC browser, zooming out 3x at least so we can see a little of the surrounding area, and selecting just a few tracks. 
```{r}
shpk_track <- import(paste0(proj_dir,"/Data/shpk.bed"))
shpk_track

session <- browserSession("UCSC")
genome(session) <- "hg19"
track(session, "SHPK_variants") <- shpk_track

shpk_snp1 <- shpk_track[1]
shpk_snp1

trackNames(session)

view <- browserView(session, 
        shpk_snp1* -3, 
        track=c("ct_SHPKvariants_7561", "ruler", "knownGene", "snp150"))
```

Note: you can also zoom in/out and select tracks from within the UCSC browser window until you get the view that you like, then you can update the code above.  
